typescript/test/wallet.test.ts

```diff
@@ -0,0 +1,246 @@
+          "00140000000000000000000000000000000000000001": 100000,
+          "76a914e6f9d74726b19b75f16fe1e9feaec048aa4fa1d088ac": 200000, // wallet legacy output
+        }
+      ),
+    ]
+
+    let bridge: MockBridge
+    let bitcoinClient: MockBitcoinClient
+    let bitcoinNetwork: BitcoinNetwork
+
+    beforeEach(async () => {
+      bridge = new MockBridge()
+      bitcoinClient = new MockBitcoinClient()
+    })
+
+    context("when wallet main UTXO is not set in the Bridge", () => {
+      beforeEach(async () => {
+        bridge.setWallet(walletPublicKeyHash.toPrefixedString(), {
+          mainUtxoHash: Hex.from(
+            "0x0000000000000000000000000000000000000000000000000000000000000000"
+          ),
+        } as Wallet)
+      })
+
+      it("should return undefined", async () => {
+        const mainUtxo = await determineWalletMainUtxo(
+          walletPublicKeyHash,
+          bridge,
+          bitcoinClient,
+          bitcoinNetwork
+        )
+
+        expect(mainUtxo).to.be.undefined
+      })
+    })
+
+    context("when wallet main UTXO is set in the Bridge", () => {
+      const tests = [
+        {
+          testName: "recent witness transaction",
+          // Set the main UTXO hash based on the latest transaction from walletWitnessTransactionHistory.
+          actualMainUtxo: {
+            transactionHash: Hex.from(
+              "ea374ab6842723c647c3fc0ab281ca0641eaa768576cf9df695ca5b827140214"
+            ),
+            outputIndex: 1,
+            value: BigNumber.from(200000),
+          },
+          expectedMainUtxo: {
+            transactionHash: Hex.from(
+              "ea374ab6842723c647c3fc0ab281ca0641eaa768576cf9df695ca5b827140214"
+            ),
+            outputIndex: 1,
+            value: BigNumber.from(200000),
+          },
+        },
+        {
+          testName: "recent legacy transaction",
+          // Set the main UTXO hash based on the second last transaction from walletLegacyTransactionHistory.
+          actualMainUtxo: {
+            transactionHash: Hex.from(
+              "05dabb0291c0a6aa522de5ded5cb6d14ee2159e7ff109d3ef0f21de128b56b94"
+            ),
+            outputIndex: 0,
+            value: BigNumber.from(100000),
+          },
+          expectedMainUtxo: {
+            transactionHash: Hex.from(
+              "05dabb0291c0a6aa522de5ded5cb6d14ee2159e7ff109d3ef0f21de128b56b94"
+            ),
+            outputIndex: 0,
+            value: BigNumber.from(100000),
+          },
+        },
+        {
+          testName: "old witness transaction",
+          // Set the main UTXO hash based on the oldest transaction from walletWitnessTransactionHistory.
+          actualMainUtxo: {
+            transactionHash: Hex.from(
+              "3ca4ae3f8ee3b48949192bc7a146c8d9862267816258c85e02a44678364551e1"
+            ),
+            outputIndex: 0,
+            value: BigNumber.from(100000),
+          },
+          expectedMainUtxo: undefined,
+        },
```

Comment:
We missed the test case for an old legacy transaction.

Reply:
See f2e5b9f9